var/spack/repos/builtin/packages/siesta/package.py

```diff
@@ -30,17 +52,44 @@ class Siesta(MakefilePackage):
         url="http://departments.icmab.es/leem/siesta/CodeAccess/Code/siesta-3.2-pl-5.tgz",
     )
 
-    depends_on("c", type="build")  # generated
-    depends_on("fortran", type="build")  # generated
-
     patch("configure.patch", when="@:4.0")
 
-    depends_on("mpi")
+    variant("mpi", default=True, description="Builds with mpi support")
+    variant("openmp", default=True, description="Enables OpenMP support")
+    variant("netcdf", default=False, description="Compile with Netcdf")
+    variant("metis", default=False, description="Activate Metis as a possible ordering library")
+    variant("elpa", default=False, description="Use ELPA")
+    variant("mumps", default=False, description="Compile with support for MUMPS solvers")
+    variant("pexsi", default=False, description="Compile with PEXSI")
+    variant(
+        "cray",
+        default=False,
+        description="Enable specific cray settings for using cray-hdf5"
+        " and cray-netcdf modulefiles",
+    )
+    variant("debug", default=False, description="Build in debug  mode")
+    variant(
+        "build_type",
+        default="Release",
+        description="The build type to build",
+        values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel", "check"),
+        when="@5.0.0: build_system=cmake",
+    )
+
+    depends_on("mpi", when="+mpi")
     depends_on("blas")
     depends_on("lapack")
     depends_on("scalapack")
     depends_on("netcdf-c")
     depends_on("netcdf-fortran")
+    depends_on("cray-libsci+openmp", when="^[virtuals=cray-libsci] cray-libsci")
+    depends_on("metis@5:", when="+metis")
+    depends_on("elpa", when="+elpa")
+    depends_on("mumps", when="+mumps")
+    depends_on("pexsi", when="+pexsi")
+
+    with when("build_system=cmake"):
+        depends_on("scalapack", when="+mpi")
```

Comment:
There already is an unconditional `depends_on("scalapack")` above, and of course `scalapack` is needed for `+mpi` in general, independent on `cmake` or `makefile`, the `depends_on("scalapack")` above should be changed to:
```suggestion
    depends_on("scalapack", when="+mpi")
```

Reply:
Good, I submitted https://github.com/CINES-HPC/spack/pull/1 with this change to be merged into your PR.